src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3.cs

```diff
@@ -368,8 +368,8 @@ public static Vector3 Cross(Vector3 vector1, Vector3 vector2)
             // This implementation is based on the DirectX Math Library XMVector3Cross method
             // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
 
-            Vector128<float> v1 = vector1.AsVector128();
-            Vector128<float> v2 = vector2.AsVector128();
+            Vector128<float> v1 = vector1.AsVector128Unsafe();
+            Vector128<float> v2 = vector2.AsVector128Unsafe();
```

Comment:
If you're going to do this, you should modify the `Shuffle` below to use `-1`, rather than `3`, for selecting `W`

It avoids a potential pessimization on some hardware if `W` is `NaN` or similar and is "free".